tree/dataframe/src/RMetaData.cxx

```diff
@@ -0,0 +1,107 @@
+// Author: Ivan Kabadzhov CERN  10/2022
+
+/*************************************************************************
+ * Copyright (C) 1995-2022, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#include "ROOT/RDF/RMetaData.hxx"
+#include <stdexcept> // std::logic_error
+
+#include <iostream>
+
+namespace ROOT {
+namespace RDF {
+namespace Experimental {
+
+void RMetaData::AddMetaData(const std::string &cat, int val)
+{
+   fJson[cat] = val;
+}
+
+void RMetaData::AddMetaData(const std::string &cat, double val)
+{
+   fJson[cat] = val;
+}
+
+void RMetaData::AddMetaData(const std::string &cat, const std::string &val)
+{
+   fJson[cat] = val;
+}
+
+void RMetaData::AddJSONMetaData(const std::string &cat, const nlohmann::json &val)
+{
+   fJson[cat] = val;
+}
+
+std::string RMetaData::Dump(const std::string &cat) const
```

Comment:
in all methods, I don't understand what `cat` stands for, should this be `key`?

Reply:
`cat` stood for category, but I totaly agree that the `key` is better